temporian/beam/io/tensorflow.py

```diff
@@ -0,0 +1,273 @@
+            if feature_schema.dtype in [
+                DType.BOOLEAN,
+                DType.INT32,
+                DType.INT64,
+            ]:
+                f(ex, feature_schema.name).int64_list.value[
+                    :
+                ] = src_value = dict_example[feature_schema.name]
+
+            elif feature_schema.dtype in [
+                DType.FLOAT32,
+                DType.FLOAT64,
+            ]:
+                f(ex, feature_schema.name).float_list.value[
+                    :
+                ] = src_value = dict_example[feature_schema.name]
+
+            elif feature_schema.dtype == DType.STRING:
+                f(ex, feature_schema.name).bytes_list.value[
+                    :
+                ] = src_value = dict_example[feature_schema.name]
+
+            else:
+                raise ValueError("Non supported feature dtype")
+
+        # Indexes
+        for index_schema in self._schema.indexes:
+            src_value = dict_example[index_schema.name]
+            if index_schema.dtype in [
+                DType.BOOLEAN,
+                DType.INT32,
+                DType.INT64,
+            ]:
+                f(ex, index_schema.name).int64_list.value.append(src_value)
+            elif index_schema.dtype in [
+                DType.FLOAT32,
+                DType.FLOAT64,
+            ]:
+                f(ex, index_schema.name).float_list.value.append(src_value)
+            elif index_schema.dtype == DType.STRING:
+                f(ex, index_schema.name).bytes_list.value.append(src_value)
+            else:
+                raise ValueError("Non supported index dtype")
+
+        yield ex
+
+
+@beam.ptransform_fn
+def from_tensorflow_record(
+    pipe,
+    file_pattern: str,
+    schema: Schema,
+    timestamp_key: str = "timestamp",
+    grouped_by_index: bool = True,
+) -> PEventSet:
+    """Imports an EventSet from a TF.Records of TF.Examples.
+
+    TF.Records of TF.Examples is one of the standard solution to store data
+    for TensorFlow models.
+    https://www.tensorflow.org/tutorials/load_data/tfrecord
+
+    The GZIP compression is used.
+
+    Usage example:
+
+    ```
+    input_node: tp.EventSetNode = ...
+    ( p
+      | tpb.from_tensorflow_record("/input.tfr.gzip", input_node.schema)
+      | ... # processing
+      | tpb.to_tensorflow_record("/output.tfr.gzip", output_node.schema)
+    )
+    ```
+
+    Args:
+        pipe: Beam pipe.
+        file_pattern: Path or path matching expression compatible with
+            `MatchFiles`.
+        schema: Schema of the data. If you have a Temporian node, the schema is
+            available with `node.schema`.
+        timestamp_key: Key containing the timestamps.
+        grouped_by_index: Are events groupped by index. Run
+            `tp.help.grouped_by_index()` for the documentation. Currently, only
+            grouped_by_index=True is implemented.
+
+    Returns:
+        A PCollection of event-set compatible with tpb.run.
```

Comment:
s/event-set/EventSet 🙏🏼

Reply:
But there is an issue here.
It is not a PCollection of EventSet object (instead it is a collection of IndexValue), which together represent an EventSet. Same as for the index, we need to normalize the name.
*Adding to tomorrow brainstorming session agenda